### PR TITLE
17633 split screen does not work without permission

### PR DESCRIPTION
**Pull request by hschink**
https://community.openproject.org/work_packages/17633

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-17633-split-screen-does-not-work-without-permission.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

**Comment by ulferts** [comment]
@hschink, please rebase.

File: frontend/app/services/work-package-service.js [hunk 1]
```diff
@@ -126,17 +135,26 @@ module.exports = function($http, PathHelper, WorkPackagesHelper, HALAPIResource,
...
-      }, force: true};
-      return workPackage.links.update.fetch(options).then(function(form) {
-        workPackage.form = form;
-        return form;
-      });
+      var modelName = 'work_package' + workPackage.id;
+
+      AuthorisationService.initModelAuth(modelName, workPackage.links);
+
+      if (AuthorisationService.can(modelName, 'update')) {
```
Comment (ulferts) [inline frontend/app/services/work-package-service.js]: IMO the authorisation should be placed in it's own method. That encapsulates the responsibilities better. 

This method would then only have to ask e.g.
`authorized_for(workPackage, 'update')`

**Comment by ulferts** [comment]
There is another bug when the user does not have the edit permission which is that for such a user, the rawDescription is rendered:

image

Other than that, and my tiny comment regarding the code, it is looking good.

**Comment by hschink** [comment]
@ulferts the bug https://github.com/opf/openproject/pull/2306#issuecomment-65634885 is handled by a16148c.

File: frontend/public/templates/components/inplace_editor.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<span ng-if="!isEditable">{{ entity.props[attribute] }}</span>
+<span ng-if="!isEditable" ng-bind-html-unsafe="readValue"></span>
```
Comment (myabc) [inline frontend/public/templates/components/inplace_editor.html]: @hschink `ng-bind-html-unsafe` is deprecated and is removed in 1.3 AFAIK.

**Comment by ulferts** [comment]
`ng-bind-html-unsafe` does already not seem to work.

With edit_work_package permission:
image

Without edit_work_packages permission:
image

**Comment by hschink** [comment]
@ulferts @myabc I squashed the fix for https://github.com/opf/openproject/pull/2306#issuecomment-65647599 and https://github.com/opf/openproject/pull/2306#discussion_r21309860 into 2b2185f.

**Pull request merged**